server/runner/taskrun/scheduler.go

```diff
@@ -984,17 +983,32 @@ func (s *Scheduler) PatchTaskStatus(ctx context.Context, task *api.Task, taskSta
 			if err := s.applicationRunner.CancelExternalApproval(ctx, issue.ID, api.ExternalApprovalCancelReasonNoTaskPendingApproval); err != nil {
 				log.Error("failed to cancel external approval on stage tasks completion", zap.Int("issue_id", issue.ID), zap.Error(err))
 			}
+		}
+	}
 
-			if issue.Project.WorkflowType == api.UIWorkflow {
-				needAttention := false
-				patch := &api.IssuePatch{
-					ID:                    issue.ID,
-					UpdaterID:             api.SystemBotID,
-					AssigneeNeedAttention: &needAttention,
-				}
-				if _, err := s.store.PatchIssue(ctx, patch); err != nil {
-					return nil, errors.Wrapf(err, "failed to patch issue assigneeNeedAttention after completing the whole stage, issuePatch: %+v", patch)
+	// If there isn't a pendingApproval task, then we need to set issue.AssigneeNeedAttention to false for UI workflow.
+	if taskPatched.Status == api.TaskPending && issue != nil && issue.Project.WorkflowType == api.UIWorkflow {
```

Comment:
This is okay for now, but we need to think of a better/scalable way to fix this e.g. without loading the entire pipeline.